diff --git a/src/wutil/gettext.rs b/src/wutil/gettext.rs
index ab968ed34..77a70f6b5 100644
--- a/src/wutil/gettext.rs
+++ b/src/wutil/gettext.rs
@@ -1,4 +1,4 @@
-use std::collections::HashMap;
+use std::{collections::HashMap, ffi::CStr};
 use std::ffi::CString;
 use std::sync::Mutex;
 
@@ -48,7 +48,7 @@ use internal::*;
 // Really init wgettext.
 fn wgettext_really_init() {
     let package_name = CString::new(PACKAGE_NAME).unwrap();
-    let localedir = CString::new(env!("LOCALEDIR")).unwrap();
+    let localedir = CStr::from_bytes_until_nul(env!("LOCALEDIR").as_bytes()).unwrap();
     fish_bindtextdomain(&package_name, &localedir);
     fish_textdomain(&package_name);
 }
